### ISSUE TITLE
traceability of an edge in KG2C to the KG2 edge it derives from

### ISSUE THREAD

**Issue by saramsey** [RTXteam/RTX#1342]
Currently, is there a way to quickly and unambiguously determine, for a given edge in KG2C, which edge in KG2 it derives from?  If not, I think we should add this information into KG2C. As we increasingly use KG2C for ARS-facing functionality, we will increasingly want to be able to "trace" where a KG2C edge comes from, back to the source KG2 and then to the upstream source.

**Comment by saramsey** [comment]
I believe we can use the edge `id` field for this.

**Comment by amykglen** [comment]
seems like a good idea! yeah, we could add an array property to KG2c edges that contains the IDs of the KG2 edges they were created from (since a single KG2c edge is the result of merging all KG2 edges with the same predicate, preferred subject curie, and preferred object curie). certainly would make tracing in KG2 easier. (although I suppose there's still a _little_ ambiguity because of the merging of edges.)

**Comment by saramsey** [comment]
Cool, thanks. I like the idea of adding a list-type property to edges in KG2C. It can contain the KG2 edge `id` property values (which of course, we would also have to define within KG2, LOL).

**Comment by saramsey** [comment]
LOL somehow I might have edited *your* comment. LOL I didn't even know that was possible in GitHub. More coffee needed.

**Comment by saramsey** [comment]
Tagging @kvarforl and @ericawood to get their take on auto-generation if edge `id` properties in KG2 (in `kg2-simplified.json`, not just in the KGX export).

**Comment by edeutsch** [comment]
If two edges with the same predicate is present between two concept clusters in KG2, what source information is retained on the final single edge in KG2C?

**Comment by amykglen** [comment]
all of it. the `provided_by` of the merged KG2 edges are combined into one list. same for `publications`.
```
{
  "predicate": "biolink:treats",
  "provided_by": [
    "MyChem:",
    "REPODB:",
    "SEMMEDDB:",
    "identifiers_org_registry:chembl.compound"
  ],
  "id": "7150133",
  "subject": "CHEMBL.COMPOUND:CHEMBL79",
  "object": "OMIM:MTHU037280",
  "publications": [
    "PMID:16194649",
    "PMID:30712496",
    "PMID:3905101",
    "PMID:4118837",
    "PMID:831109",
    "PMID:9217675"
  ]
}
```

**Comment by edeutsch** [comment]
fantastic!

**Comment by kvarforl** [comment]
Yep, I think it's feasible! I'll touch base with Erica so we can get on the same page of where in the build process is best to add the ids. 
If curious, discussion is here: https://github.com/RTXteam/RTX/issues/1245
